packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx

```
@@ -0,0 +1,224 @@
+
+  const form = useForm<ForgotPasswordModalForm>({
+    resolver: zodResolver(isLastStep ? schema : schema.pick({ [currentStep]: true })),
+  })
+  const { authModalOpenName, setAuthModalOpenName } = useAuthStore(
+    (state) => ({
+      authModalOpenName: state.authModalOpenName,
+      setAuthModalOpenName: state.actions.setAuthModalOpenName,
+    }),
+    shallow
+  )
+  const handleEmailAndSeedStepSubmit = async (data: ForgotPasswordModalForm) => {
+    setIsLoading(true)
+
+    try {
+      const time = Date.now() - 10_000
+
+      const keypair = keyring.addFromMnemonic(data['EmailAndSeedStep'].mnemonic)
+      const address = keypair.address
+      const loginPayload = {
+        joystreamAccountId: address,
+        gatewayName: atlasConfig.general.appName,
+        timestamp: time,
+        action: 'login' as const,
+      }
+      const loginSignature = await keypair.sign(JSON.stringify(loginPayload))
+      const loginResponse = await loginRequest(u8aToHex(loginSignature), loginPayload)
+      const accData = await lazyCurrentAccountQuery()
+
+      if (accData.data?.accountData.email !== data['EmailAndSeedStep'].email) {
+        form.setError(`${ForgotPasswordStep.EmailAndSeedStep}.email`, {
+          message: 'Provided email do not match mnemonic.',
+        })
+        logoutRequest().catch(() => undefined)
+        return
+      }
+
+      setAccountId(loginResponse.data.accountId)
+      setCurrentStep(ForgotPasswordStep.NewPasswordStep)
+    } catch (error) {
+      displaySnackbar({
+        title: 'Something went wrong',
+        description: `We encountered unexpected error. Please try again.`,
+        iconType: 'error',
+      })
+      SentryLogger.error('Failed to login when recovering password', 'ForgotPasswordModal', error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const handleNewPasswordStep = useCallback(
+    async (data: ForgotPasswordModalForm) => {
+      if (!accountId) {
+        return
```

I would throw an error here, and wrap this whole code with try-catch, not only the axios part, there are a lot of things that can go wrong here.